superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -215,3 +218,23 @@ export const getDashboard = dashboardId => async dispatch => {
     throw error;
   }
 };
+
+//  Get dashboards the slice is added to
+export const getSliceDashboards = slice => async dispatch => {
+  if (slice) {
```

Comment:
the `if` clause is redundant because the `try...cache`  handled exception

Reply:
My intention was for the `try...catch` clause to just handle request errors or other unexpected errors.  The `if` clause is there because this function might be called with no `slice` param, in the case of a chart that hasn't yet been saved, and I wanted it to return `[]` in that case as part of normal, non-exception behavior.

Reply:
if there is no `slice` param, the request will also raise an exception, so the `catch` statement alway catches an exception although an invalid input param.

Reply:
But, I don't want to send the request at all if there's no `slice` param, I just want to return an empty array.

Reply:
if you don't want to send a request, you shouldn't call this action. e.g.:
```
    const originalDashboards = this.props.slice ? await this.props.actions.getSliceDashboards(
      this.props.slice,
    ) : [];
```
just a personal suggestion, optimize it for the future.  thanks for the fixing.